src/compiler/checker.ts

```diff
@@ -18595,7 +18595,19 @@ namespace ts {
             // Grammar checking
             checkGrammarStatementInAmbientContext(node);
 
-            checkExpression(node.expression);
+            const type = checkExpression(node.expression);

```

Comment:
So, yeah, no difference between 
```ts
async function bar(){
   return isAsync();
}
// and
async function nested(){
   return await isAsync();
}

var res = await bar();
```

So, but this construction is incorrect, because second await resolve the promise value, even is there is a nested promise
```ts
await await bar();
```